app/controllers/outputs_controller.rb

```diff
@@ -0,0 +1,97 @@
+class OutputsController < ApplicationController
+  ALLOWED_RESOURCE_TYPES = %w(study_impact dissemination publication)
+
+  def create
+    @study = Study.find(params[:study_id])
+    resource_type = params[:output_type]
+    if ALLOWED_RESOURCE_TYPES.include? resource_type
+      self.send("create_#{resource_type}")
+    else
+      raise ActionController::RoutingError.new('Not Found')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.